### PR TITLE
fix: change registration form for onramp

### PR DESCRIPTION
**Pull request by ahtesham-quraish**
Description:
In this we have changed the registration form on the basis of variant query param and hide the multiple things like registration button

How Has This Been Tested?
I have tested it locally.

Its related to: 
https://2u-internal.atlassian.net/browse/VAN-1476

Screenshots
Screenshot 2023-06-13 at 11 29 12 AM

### CODE REVIEW

**Review by zainab-amir — COMMENTED** [review]
Paragon classes should be used if needed

File: src/data/constants.js [hunk 1]
```diff
@@ -35,4 +35,4 @@ export const INVALID_NAME_REGEX = /[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{
 
 // Query string parameters that can be passed to LMS to manage
 // things like auto-enrollment upon login and registration.
-export const AUTH_PARAMS = ['course_id', 'enrollment_action', 'course_mode', 'email_opt_in', 'purchase_workflow', 'next', 'save_for_later', 'register_for_free', 'track', 'is_account_recovery'];
+export const AUTH_PARAMS = ['course_id', 'enrollment_action', 'course_mode', 'email_opt_in', 'purchase_workflow', 'next', 'save_for_later', 'register_for_free', 'track', 'is_account_recovery', 'embed'];
```
[the remark this answers is not in the document]
Comment (ahtesham-quraish) [reply src/data/constants.js]: resolved

File: src/logistration/Logistration.jsx [hunk 2]
```diff
@@ -79,58 +83,75 @@ const Logistration = (props) => {
...
-
-  return (
-    <BaseComponent>
-      <div>
-        {disablePublicAccountCreation
-          ? (
-            <>
-              <Redirect to={updatePathWithQueryParams(LOGIN_PAGE)} />
-              {institutionLogin && (
+  const renderForm = () => (
```
Comment (zainab-amir) [inline src/logistration/Logistration.jsx:86]:
```suggestion
  const renderLogistrationTabs = () => (
```

File: src/logistration/Logistration.jsx [hunk 3]
```diff
@@ -79,58 +83,75 @@ const Logistration = (props) => {
...
+        : (
+          <div>
+            {institutionLogin
+              ? (
                 <Tabs defaultActiveKey="" id="controlled-tab" onSelect={handleInstitutionLogin}>
-                  <Tab title={tabTitle} eventKey={LOGIN_PAGE} />
+                  <Tab title={tabTitle} eventKey={selectedPage === LOGIN_PAGE ? LOGIN_PAGE : REGISTER_PAGE} />
                 </Tabs>
+              )
+              : (!variant && !isValidTpaHint() && (
```
Comment (zainab-amir) [inline src/logistration/Logistration.jsx:113]:
```suggestion
              : (!isRegistrationEmbedded && !isValidTpaHint() && (
```

File: src/logistration/Logistration.jsx [hunk 4]
```diff
@@ -79,58 +83,75 @@ const Logistration = (props) => {
...
+                </Tabs>
+              ))}
+            { key && (
+              <Redirect to={updatePathWithQueryParams(key)} />
+            )}
+            <div
+              id="main-content"
+              className={classNames(
+                'main-content',
+                { 'main-content-background': variant },
```
Comment (zainab-amir) [inline src/logistration/Logistration.jsx:126]: This class needs to be removed.

File: src/logistration/Logistration.jsx [hunk 5]
```diff
@@ -79,58 +83,75 @@ const Logistration = (props) => {
...
+          { renderForm() }
+        </BaseComponent>
+      )}
+      {variant && (
+        <>
+          { renderForm() }
+        </>
+      )}
+    </>
   );
```
Comment (zainab-amir) [inline src/logistration/Logistration.jsx:155]:
```suggestion
  return (
    isRegistrationEmbedded ? (
      renderForm()
    ) : (
      <BaseComponent>
        {renderForm()}
      </BaseComponent>
    )
  );
```

File: src/logistration/Logistration.jsx [hunk 6]
```diff
@@ -36,6 +39,7 @@ const Logistration = (props) => {
   const [institutionLogin, setInstitutionLogin] = useState(false);
   const [key, setKey] = useState('');
   const disablePublicAccountCreation = getConfig().ALLOW_PUBLIC_ACCOUNT_CREATION === false;
+  const { variant } = getAllPossibleQueryParams();
```
Comment (zainab-amir) [inline src/logistration/Logistration.jsx:42]:
```suggestion
  const isRegistrationEmbedded = getAllPossibleQueryParams()?.variant === 'embedded' && selectedPage === REGISTER_PAGE;
```

File: src/register/RegistrationPage.jsx [hunk 7]
```diff
@@ -86,6 +87,7 @@ const RegistrationPage = (props) => {
   const { formatMessage } = useIntl();
   const countryList = useMemo(() => getCountryList(getLocale()), []);
   const queryParams = useMemo(() => getAllPossibleQueryParams(), []);
+  const { variant } = queryParams;
```
Comment (zainab-amir) [inline src/register/RegistrationPage.jsx:90]: Similar approach can be used as above. 
```suggestion
  const isRegistrationEmbedded = queryParams?.variant === 'embedded';
```

File: src/register/RegistrationPage.jsx [hunk 8]
```diff
@@ -527,7 +529,12 @@ const RegistrationPage = (props) => {
             <Spinner animation="border" variant="primary" id="tpa-spinner" />
           </div>
         ) : (
-          <div className="mw-xs mt-3">
+          <div
+            className={classNames(
+              'mw-xs mt-3',
+              { 'form--centerd': variant },
```
Comment (zainab-amir) [inline src/register/RegistrationPage.jsx:535]: We can use paragon classes for this
```suggestion
              { 'w-100 m-auto': isRegistrationEmbedded },
```

File: src/register/RegistrationPage.jsx [hunk 9]
```diff
@@ -611,13 +618,15 @@ const RegistrationPage = (props) => {
...
                 onMouseDown={(e) => e.preventDefault()}
               />
-              <ThirdPartyAuth
-                currentProvider={currentProvider}
-                providers={providers}
-                secondaryProviders={secondaryProviders}
-                handleInstitutionLogin={handleInstitutionLogin}
-                thirdPartyAuthApiStatus={thirdPartyAuthApiStatus}
-              />
+              {!variant && (
```
Comment (zainab-amir) [inline src/register/RegistrationPage.jsx:621]:
```suggestion
              {!isRegistrationEmbedded && (
```

**Review by zainab-amir — CHANGES_REQUESTED** [review]

**Review by zainab-amir — APPROVED** [review]

**Pull request merged**